File: day-01-binary-search/python/max-guesses.py
import math

def max_guesses(n):
    if n < 0:
        raise ValueError("List size cannot be negative")
    if n == 0:
        return 0
    if n == 1:
        return 1
    return math.ceil(math.log2(n))

print(max_guesses(1024))
print(max_guesses(100))
print(max_guesses(104545))
print(max_guesses(1))
print(max_guesses(0))
# print(max_guesses(-1)) raise ValueError

File: day-01-binary-search/js/max-guesses.js
function maxGuesses(n) {
  if (n < 0) return undefined;
  if (n === 0) return 0;
  if (n === 1) return 1;
  return Math.ceil(Math.log2(n));
}

console.log(maxGuesses(1024));
console.log(maxGuesses(100));
console.log(maxGuesses(0));
console.log(maxGuesses(-1));
